api/buckets/service.go

```diff
@@ -2084,26 +2147,54 @@ func (s *Service) Archive(ctx context.Context, req *pb.ArchiveRequest) (*pb.Arch
 		}
 	}
 
+	ba, err := s.Collections.BucketArchives.GetOrCreate(ctx, req.Key)
+	if err != nil {
+		return nil, fmt.Errorf("getting bucket archive data: %s", err)
+	}
+
+	var archiveConfig *mdb.ArchiveConfig
+	if req.ArchiveConfig != nil {
+		archiveConfig = fromPbArchiveConfig(req.ArchiveConfig)
+	} else if ba.DefaultArchiveConfig != nil {
+		archiveConfig = ba.DefaultArchiveConfig
+	} else {
+		// The queried BucketArchive was created before we added .DefaultArchiveConfig so just use the default default
+		// and defer saving the default until the user calls DefaultArchiveConfic or SetDefaultArchiveConfig.
+		archiveConfig = &defaultDefaultArchiveConfig
+	}
+
+	storageConfig := baseArchiveStorageConfig
+	storageConfig.Cold.Filecoin = toFilConfig(archiveConfig)
+
+	if storageConfig.Cold.Filecoin.Addr == "" {
+		// We don't have an address to use, which is the case for a BucketArchive.DefaultArchiveConfig
+		// that is the default value, get the default address from the FFS instance
+		storageConfig.Cold.Filecoin.Addr = defConf.Cold.Filecoin.Addr
```

Comment:
The default `ArchiveConfig` doesn't include any address information, and if the user sets their own Bucket default `ArchiveConfig`, it may not have an address. In that case, grab the default address from the FFS instance. This is nice because it can change if the user wants. 

Long term, I'm working on a Powergate branch where default address in managed separately from default config, and address can be provided as an option in `PushConfig`, `Replace`, etc. That will make this cleaner.